lua/hawtkeys/ui.lua

```diff
@@ -138,6 +114,48 @@ M.search = function(text)
                 0,
                 -1
             )
+        else
+            local newStr = text:lower()
+            local marked = {} -- Keep track of characters already marked
+            for idx = 1, #data.combo do
+                local char = data.combo:sub(idx, idx)
+                local pos = marked[char] or 1 -- Start searching from the last marked position or from the beginning
+                pos = newStr:find(char, pos, true) or 0
+                if marked[char] and marked[char] == pos then
+                    pos = newStr:find(char, pos + 1, true) or 0
+                end
+                if pos then
+                    newStr = newStr:sub(1, pos - 1)
+                        .. char
+                        .. newStr:sub(pos + 1)
+                    local hl_col = pos + 5
+                    vim.api.nvim_buf_add_highlight(
+                        ResultBuf,
+                        -1,
+                        "Function",
+                        line_count - (already_mapped and 3 or 1),
+                        hl_col - 1,
+                        hl_col
+                    )
+                    marked[char] = pos -- Mark this character's position
+                end
+            end
+
+            local score_offset = #line - #tostring(data.score)
+            local score_hl = "MoreMsg"
+            if data.score <= 0 then
+                score_hl = "ErrorMsg"
+            elseif data.score <= 2 then
```

Comment:
Can we add a if =>6 then a SucessMsg?

Reply:
Also, instead of hardcoding the scores. Can you define a stable at the top for low score, midscore and highscore?

Reply:
> Can we add a if =>6 then a SucessMsg?

MoreMsg is green for me, SuccessMsg is not a default highlight group as far as I know.

Reply:
I added an underline with a custom hlgroup for matches >=6